lib/rgeo/cartesian/planar_graph.rb

```diff
@@ -0,0 +1,400 @@
+# frozen_string_literal: true
+
+module RGeo
+  module Cartesian
+    module Graphs
```

Comment:
Maybe a different name?

Reply:
Why not `PlanarGraph` (as suggested by the file). Other classes could be subclasses. Except for `GeometryGraph` IMHO

Reply:
Sorry, I'm not sure exactly what you mean. Would you mind clarifying? I'll do this last since it will create a lot of diffs.

Reply:
I mean having the `PlanarGraph` class directly be parent of `HalfEdge`. Hence you would have:

- `RGeo::Cartesian::PlanarGraph`
- `RGeo::Cartesian::PlanarGraph::HalfEdge`
- `RGeo::Cartesian::GeometryGraph`